indico/modules/categories/util.py

```diff
@@ -138,6 +139,52 @@ def serialize_category_atom(category, url, user, event_filter):
     return BytesIO(feed.to_string().encode('utf-8'))
 
 
+def serialize_category(category, with_favorite=False, with_path=False, parent_path=None, child_path=None):
+    data = {
+        'id': category.id,
+        'title': category.title,
+        'is_protected': category.is_protected,
+        'has_events': category.has_events,
+        'deep_category_count': category.deep_children_count,
+        'deep_event_count': category.deep_events_count,
+        'can_access': category.can_access(session.user),
+    }
+    if with_path:
+        if child_path:
+            data['path'] = child_path[:]
+            for segment in reversed(child_path):
+                data['path'].pop()
+                if not data['path'] or data['path'][-1]['id'] == category.id:
+                    break
+        elif parent_path:
+            data['path'] = parent_path[:]
+            data['path'].append({'id': category.id, 'title': category.title})
+        else:
+            data['path'] = [{'id': c.id, 'title': c.title}
+                            for c in category.chain_query.options(load_only('id', 'title'))]
+        data['parent_path'] = data['path'][:-1]
+    if with_favorite:
+        data['is_favorite'] = session.user and category in session.user.favorite_categories
+    return data
+
+
+def serialize_category_chain(category, include_children=False, include_parents=False):
+    data = {'category': serialize_category(category, with_path=True)}
+    if include_children:
+        data['subcategories'] = [serialize_category(c, with_path=True, parent_path=data['category']['path'])
+                                 for c in category.children]
+    if include_parents:
+        data['supercategories'] = [serialize_category(c, with_path=True, child_path=data['category']['path'])
+                                   for c in category.parent_chain_query]
+    return data
+
+
+def serialize_category_chain_by_id(id, include_children=False, include_parents=False):
```

Comment:
But what's the problem with getting the category object outside and passing it to `serialize_category_chain`?